### ISSUE TITLE
ISODateFromFields has:

### ISSUE THREAD

**Issue by Ms2ger** [tc39/proposal-temporal#2099]

**Comment by ptomato** [comment]
Good point, I guess there are a couple of cases where it would be observable:
- `calendar.dateFromFields({ get year() { throw "not called"; } })` (the one you already mentioned)
- `calendar.dateFromFields({ year: 2022, monthCode: "not valid" })` (would throw TypeError instead of RangeError)
